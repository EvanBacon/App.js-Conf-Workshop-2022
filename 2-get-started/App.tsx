import { StyleSheet, View } from "react-native";
import { Step1 } from "./step-1";
// import { Step2 } from "./step-2";
// import { Step3 } from "./step-3";
// import { Step4 } from "./step-4";
// import { Step5 } from "./step-5";
// import { Step6 } from "./step-6";
// import { Step7 } from "./step-7";

export default function App() {
  return (
    <View style={styles.container}>
      <Step1 />
      {/* <Step2 /> */}
      {/* <Step3 /> */}
      {/* <Step4 /> */}
      {/* <Step5 /> */}
      {/* <Step6 /> */}
      {/* <Step7 /> */}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
});
