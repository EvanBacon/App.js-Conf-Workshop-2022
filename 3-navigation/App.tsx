
import { Step1 } from "./step-1";
// import { Step2 } from "./step-2";
// import { Step3 } from "./step-3";
// import { Step4 } from "./step-4";
// import { Step5 } from "./step-5";

export default function App() {
  return (
    <>
      <Step1 />
      {/* <Step2 /> */}
      {/* <Step3 /> */}
      {/* <Step4 /> */}
      {/* <Step5 /> */}
    </>
  );
}
