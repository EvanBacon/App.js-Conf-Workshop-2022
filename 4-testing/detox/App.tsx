import React, { useState } from "react";
import { StyleSheet, Text, View, Button } from "react-native";

export default function App() {
  const [message, setMessage] = useState("");
  return (
    <View style={styles.container}>
      <Text testID="welcome">Welcome!</Text>
      <Button
        onPress={() => setMessage("Hello!!!")}
        title="Say hello"
        testID="hello_button"
      />
      <Text>{message}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
});
