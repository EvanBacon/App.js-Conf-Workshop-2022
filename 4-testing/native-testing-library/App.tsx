import React from "react";
import { StyleSheet, Text, View } from "react-native";

export default function App() {
  return (
    <View style={styles.container}>
      <Text>React Europe 2020!</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
});
