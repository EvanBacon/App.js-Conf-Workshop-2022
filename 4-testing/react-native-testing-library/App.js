import React, { useState } from "react";
import { Button, StyleSheet, Text, View, TextInput } from "react-native";

function App(props) {
  const { onCountIncreased } = props;
  const [count, setCount] = useState(0);
  const [inputCount, setInputCount] = useState(0);

  function handleAddOne() {
    setCount(count + 1);
    if (onCountIncreased) onCountIncreased(count + 1);
  }

  function handleAddMore() {
    if (isNaN(inputCount)) return;

    setCount(inputCount + count);
    if (onCountIncreased) onCountIncreased(inputCount + count);
  }

  return (
    <View style={styles.container}>
      <Text testID="counter" style={styles.count}>
        {count}
      </Text>
      <Button title="Add one" onPress={() => handleAddOne()} />
      <TextInput
        testID="text-input"
        onChangeText={(text) => setInputCount(Number(text))}
        style={styles.input}
      />
      <Button title="Add more" onPress={() => handleAddMore()} />
    </View>
  );
}

export default App;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
  input: {
    padding: 16,
    borderWidth: 1,
    borderColor: "#888",
    marginHorizontal: 16,
  },
  count: {
    fontSize: 20,
    marginBottom: 32,
    fontWeight: "bold",
  },
});
