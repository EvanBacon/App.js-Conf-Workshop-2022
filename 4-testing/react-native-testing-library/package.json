{
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "test": "jest"
  },
  "dependencies": {
    "expo": "^37.0.0",
    "react": "16.9.0",
    "react-dom": "16.13.1",
    "react-native": "https://github.com/expo/react-native/archive/sdk-37.0.1.tar.gz",
    "react-native-gesture-handler": "~1.6.0",
    "react-native-web": "0.12.2",
    "react-navigation": "^3.11.0"
  },
  "devDependencies": {
    "babel-preset-expo": "^8.1.0",
    "jest-expo": "37.0.0",
    "react-native-testing-library": "~1.7.0"
  },
  "jest": {
    "preset": "jest-expo",
    "setupFiles": [
      "./jest/setup"
    ]
  },
  "private": true
}
