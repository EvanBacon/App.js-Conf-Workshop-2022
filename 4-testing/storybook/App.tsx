console.disableYellowBox = true;

export default from './storybook';
