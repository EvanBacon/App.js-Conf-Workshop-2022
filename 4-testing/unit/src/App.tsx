import React from "react";
import { StyleSheet, Text, View, Platform } from "react-native";

export default function App() {
    return (
        <View style={styles.container}>
            <Text>Open up App.tsx to start working on your app!</Text>
            {Platform.OS === "ios" && <Text>I'm running on iOS!</Text>}
            {Platform.OS === "web" && <Text>I'm running on web!</Text>}
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: "#fff",
        alignItems: "center",
        justifyContent: "center",
    },
});
