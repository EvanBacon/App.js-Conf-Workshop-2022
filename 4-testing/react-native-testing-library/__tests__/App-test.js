import * as React from "react";
import { render, fireEvent } from "react-native-testing-library";
import App, { incrementCount } from "../App";

// Snapshot test
test("should render", () => {
  expect(render(<App />).toJSON()).toMatchSnapshot();
});

// Interact with a rendered component
test("should increase count on click", () => {
  const { getByTestId, getByText } = render(<App />);
  expect(getByTestId("counter").props.children).toEqual(0);
  fireEvent.press(getByText("Add one"));
  expect(getByTestId("counter").props.children).toEqual(1);
});

// Interact with an input
test("should increase count on input", () => {
  const { getByTestId, getByText } = render(<App />);
  expect(getByTestId("counter").props.children).toEqual(0);

  fireEvent.changeText(getByTestId("text-input"), 5);
  fireEvent.press(getByText("Add more"));

  expect(getByTestId("counter").props.children).toEqual(5);
});

// Test both work together
test("should increase count on input and button press", () => {
  const { getByTestId, getByText } = render(<App />);
  expect(getByTestId("counter").props.children).toEqual(0);

  fireEvent.changeText(getByTestId("text-input"), 5);
  fireEvent.press(getByText("Add more"));
  fireEvent.press(getByText("Add one"));

  expect(getByTestId("counter").props.children).toEqual(6);
});

// Use mocks to test that props are invoked correctly
test("should fire onCountIncreased callback", () => {
  let onCountIncreased = jest.fn();
  const { getByTestId, getByText } = render(
    <App onCountIncreased={onCountIncreased} />
  );
  fireEvent.press(getByText("Add one"));
  fireEvent.press(getByText("Add one"));
  expect(onCountIncreased).toHaveBeenCalledTimes(2);
  expect(onCountIncreased).toHaveBeenNthCalledWith(1, 1);
  expect(onCountIncreased).toHaveBeenNthCalledWith(2, 2);
});
