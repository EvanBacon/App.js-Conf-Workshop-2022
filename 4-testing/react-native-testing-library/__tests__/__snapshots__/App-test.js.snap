// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
        "marginBottom": 32,
      }
    }
    testID="counter"
  >
    0
  </Text>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Add one
      </Text>
    </View>
  </View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#888",
        "borderWidth": 1,
        "marginHorizontal": 16,
        "padding": 16,
      }
    }
    testID="text-input"
    underlineColorAndroid="transparent"
  />
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Add more
      </Text>
    </View>
  </View>
</View>
`;
